src/providers/oauth-provider.js

```diff
@@ -11,6 +11,7 @@ var defaults = {
   clientId: null,
   clientSecret: null,
   grantPath: '/oauth2/token',
+  grantType: 'password',
```

Comment:
Hi @pierrejochem, thanks for your PR. Since `grantType` is configurable, and always requirable, it should be added to `requiredKeys` list to avoid its deletion. Could you add a test to it please?